tests/integration/test_charm.py

```diff
@@ -25,23 +26,25 @@
 AUTH_PROXY_REQUIRER = "auth-proxy-requirer"
 
 
-async def get_unit_address(ops_test: OpsTest, app_name: str, unit_num: int) -> str:
-    """Get private address of a unit."""
-    status = await ops_test.model.get_status()  # noqa: F821
-    return status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
-
-
-async def get_app_address(ops_test: OpsTest, app_name: str) -> str:
-    """Get address of an app."""
-    status = await ops_test.model.get_status()  # noqa: F821
-    return status["applications"][app_name]["public-address"]
+async def get_k8s_service_address(ops_test: OpsTest, service_name: str) -> Optional[str]:
+    """Get the address of a LoadBalancer Kubernetes service using kubectl."""
+    model = ops_test.model.info
+    try:
+        result = sh.kubectl(
+            f"-n {model.name} get service/{service_name} -o=jsonpath='{{.status.loadBalancer.ingress[0].ip}}'".split()
+        )
```

Comment:
I thought that usually we would use `lightkube` for interacting with the k8s server, any reason not to do that?